src/LinearOperators.jl

```diff
@@ -85,346 +90,355 @@ end
+  v = rand(n)
+  w = op * v
   s = dot(w, w);  # = (Av)'(Av) = v' A' A v.
-  y = op * w;
+  y = op * w
   t = dot(v, y);  # = v' A A v.
-  ε = eps(Float64);
-  return abs(s - t) < (abs(s) + ε) * ε^(1/3);
+  ε = eps(Float64)
+  return abs(s - t) < (abs(s) + ε) * ε^(1/3)
 end
 
-check_hermitian(M :: KindOfMatrix) = check_hermitian(LinearOperator(M))
+check_hermitian(M :: AbstractMatrix) = check_hermitian(LinearOperator(M))
 
 "Cheap check that the operator is positive (semi-)definite."
 function check_positive_definite(op :: AbstractLinearOperator; semi=false)
-  m, n = size(op);
-  v = rand(n);
-  w = op * v;
-  vw = dot(v, w);
-  ε = eps(Float64);
+  m, n = size(op)
+  v = rand(n)
+  w = op * v
+  vw = dot(v, w)
+  ε = eps(Float64)
   if imag(vw) > sqrt(ε) * abs(vw)
     return false
   end
-  vw = real(vw);
+  vw = real(vw)
   return semi ? (vw ≥ 0) : (vw > 0)
 end
 
-check_positive_definite(M :: KindOfMatrix) = check_positive_definite(LinearOperator(M))
+check_positive_definite(M :: AbstractMatrix) = check_positive_definite(LinearOperator(M))
 
 # Special linear operators.
 
-"Identity operator of order `n` and of data type `dtype`."
-opEye(n :: Int; dtype=Float64) = LinearOperator(n, n, dtype, true, true,
-                                                v -> v[:], u -> u[:], w -> w[:])
-
-"Operator of all ones of size `nrow`-by-`ncol` and of data type `dtype`."
-opOnes(nrow, ncol; dtype=Float64) = LinearOperator(nrow, ncol, dtype,
-                                                   nrow == ncol, nrow == ncol,
-                                                   v -> sum(v) * ones(nrow),
-                                                   u -> sum(u) * ones(ncol),
-                                                   w -> sum(w) * ones(ncol))
-
-"Zero operator of size `nrow`-by-`ncol` and of data type `dtype`."
-opZeros(nrow, ncol; dtype=Float64) = LinearOperator(nrow, ncol, dtype,
-                                                    nrow == ncol, nrow == ncol,
-                                                    v -> zeros(nrow),
-                                                    u -> zeros(ncol),
-                                                    w -> zeros(ncol))
+"Identity operator of order `n` and of data type `T`."
+opEye(T :: DataType, n :: Int) = LinearOperator{T}(n, n, true, true,
+                                                   v -> v[:], u -> u[:], w -> w[:])
+opEye(n :: Int) = opEye(Float64, n)
+
+"Operator of all ones of size `nrow`-by-`ncol` and of data type `T`."
+opOnes(T :: DataType, nrow :: Int, ncol :: Int) = LinearOperator{T}(nrow, ncol, nrow == ncol, nrow == ncol,
+                                                                    v -> sum(v) * ones(nrow),
+                                                                    u -> sum(u) * ones(ncol),
+                                                                    w -> sum(w) * ones(ncol))
+opOnes(nrow :: Int, ncol :: Int) = opOnes(Float64, nrow, ncol)
```

Comment:
I think not because Julia wouldn't know to call the `ones()` that creates an array or this one.